### ISSUE TITLE
Checkbox inconsistencies

### ISSUE THREAD

**Issue by RomanKornev** [notable/notable#292]
- **OS Version**: Windows 10 x64
- **Notable Version**: master 0a77ac33d42fcd16b4937c7f617ae514c91e5607

### Current behavior

After #169 I noticed some checkbox inconsistencies:
Apparently, Github doesn't allow no-space between brackets `- [] check` to be used as a checkmark. Only single space `- [ ] check`.
But Notable renders `- [] check` as a checkbox. As a result, clicking on it in preview mode deletes some characters on the next line and messes up the whole formatting.

Also putting tags in lists:
```
- [](@tag/tag)
```
Currently, it's a checkbox instead of a tag.

Also syntax like `- [ ]check` and  `- []check`. It doesn't work on Github but renders in Notable.

### Expected behavior

If we want to follow GitHub markdown, then checkboxes should have:
- 1 space between brackets 
- At least 1 space between the closing bracket and its name

This will also fix putting tags in lists.

For reference:
```
- [ ] check - as expected
- [x] check - as expected
- [  ] check - not displayed as expected
- [] check - shouldn't be rendered
- [x]check - shouldn't be rendered
- [ ]check - shouldn't be rendered
```
- [ ] check - as expected
- [x] check - as expected
- [  ] check - not displayed as expected
- [] check - shouldn't be rendered
- [x]check - shouldn't be rendered
- [ ]check - shouldn't be rendered

Related #19

**Comment by fabiospampinato** [comment]
Good catch, we can definitely avoid breaking links, but following GFM more closely would probably require a more customizable Markdown renderer that we currently use 🤔

**Comment by fabiospampinato** [comment]
> Also putting tags in lists:
> Currently, it's a checkbox instead of a tag.

It turns out to be an upstream issue https://github.com/showdownjs/showdown/issues/638

The other things are acceptable IMHO.

**Comment by RomanKornev** [comment]
> The other things are acceptable IMHO.

Toggling checkbox still messes up formatting though.
It would lead people to believe that `- [] check` is acceptable syntax.
Then this happenes:
![vnuupuuf4f](https://user-images.githubusercontent.com/5807561/51329650-03e8b400-1a87-11e9-863c-b5f5e64cddd5.gif)

**Comment by fabiospampinato** [comment]
@RomanKornev that shouldn't happen, can you paste the content of that note?

**Comment by fabiospampinato** [comment]
Fixed in https://github.com/fabiospampinato/notable/commit/ebbfdbb2d995ced3573887c5cd1aacf5b6beb397

Thanks for catching all these bugs ;)

**Comment by RomanKornev** [comment]
Glad I could help.

**Comment by fabiospampinato** [comment]
This will probably get fixed once we switch to marked #531

**Comment by fabiospampinato** [comment]
Fixed in v1.6.
